service/src/main/java/com/epam/esm/service/impl/CertificateServiceImpl.java

```diff
@@ -85,64 +95,80 @@ public CertificateDto update(Long id, CertificateDto entity) {
     @Transactional
     @Override
     public void remove(Long id) {
-        CertificateDto deletedCertificate = certificateDtoMapper.toDTO(certificateDao.getEntityById(id)
-                .orElseThrow(() -> new CertificateNotFoundException(id)));
+        CertificateDto deletedCertificate = certificateDao.getById(id)
+                .map(certificateDtoMapper::toDTO)
+                .orElseThrow(() -> new EntityNotFoundException("Certificate", id));
         deleteTags(deletedCertificate);
-        if (!certificateDao.removeEntity(id)) {
-            throw new CertificateNotFoundException(id);
+        deleteOrders(deletedCertificate);
+        if (!certificateDao.remove(id)) {
+            throw new EntityNotFoundException("Certificate", id);
```

Comment:
I suppose it will be different kind of exception, as you have already checked that entity exists